Content.Server/DoAfter/DoAfter.cs

```diff
@@ -162,6 +164,25 @@ private bool IsCancelled(IEntityManager entityManager)
                 }
             }
 
+            //Allow movement but enforce distance
+            if (EventArgs.BreakOnDistance != null)
+            {
+                if(EventArgs.Target != null && !EventArgs.User.Equals(EventArgs.Target))
+                {
+                    //recalculate Target location in case Target has also moved
+                    TargetGrid = entityManager.GetComponent<TransformComponent>(EventArgs.Target!.Value).Coordinates;
+                    if (!entityManager.GetComponent<TransformComponent>(EventArgs.User).Coordinates.InRange(entityManager, TargetGrid, EventArgs.BreakOnDistance!.Value))
+                        return true;
+                }
+                if (EventArgs.Used != null && !EventArgs.User.Equals(EventArgs.Used))
+                {
+                    //recalculate Used location in case Used has also moved
+                    UsedGrid = entityManager.GetComponent<TransformComponent>(EventArgs.Used!.Value).Coordinates;
+                    if (!entityManager.GetComponent<TransformComponent>(EventArgs.User).Coordinates.InRange(entityManager, UsedGrid, EventArgs.BreakOnDistance!.Value))
+                        return true;
+                }
```

Comment:
There's a lot of yucky null handling here, also I think the used distance should be irrelevant.

Reply:
We can logically remove a few of these checks, like the target shouldn't be null, and the user shouldn't ever be the used entity. Used is optional, but is needed for this instance where the user and target are the same, and the food on the floor is a separate entity we want to calculate distance from.
We can remove Used entirely if we decide we don't want to use it for cases like that, although it doesn't seem to allow force feeding in this instance.